src/main/java/se/sundsvall/supportmanagement/service/scheduler/emailreader/EmailReaderScheduler.java

```diff
@@ -1,50 +1,43 @@
 package se.sundsvall.supportmanagement.service.scheduler.emailreader;
 
-import net.javacrumbs.shedlock.spring.annotation.SchedulerLock;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
-import org.springframework.scheduling.annotation.Scheduled;
+import org.springframework.beans.factory.annotation.Value;
 import org.springframework.stereotype.Service;
-import se.sundsvall.dept44.requestid.RequestId;
+import se.sundsvall.dept44.scheduling.Dept44Scheduled;
+import se.sundsvall.dept44.scheduling.health.Dept44HealthUtility;
 
 @Service
 public class EmailReaderScheduler {
 
 	private static final Logger LOG = LoggerFactory.getLogger(EmailReaderScheduler.class);
 
 	private final EmailReaderWorker emailReaderWorker;
-	private final EmailProcessingHealthIndicator healthIndicator;
+	private final Dept44HealthUtility healthUtility;
 
-	public EmailReaderScheduler(final EmailReaderWorker emailReaderWorker, final EmailProcessingHealthIndicator healthIndicator) {
+	@Value("${scheduler.emailreader.name}")
+	private String jobName;
+
+	public EmailReaderScheduler(final EmailReaderWorker emailReaderWorker, final Dept44HealthUtility healthUtility) {
 		this.emailReaderWorker = emailReaderWorker;
-		this.healthIndicator = healthIndicator;
+		this.healthUtility = healthUtility;
 	}
 
-	@Scheduled(cron = "${scheduler.emailreader.cron}")
-	@SchedulerLock(name = "fetch_emails", lockAtMostFor = "${scheduler.emailreader.shedlock-lock-at-most-for}")
+	@Dept44Scheduled(cron = "${scheduler.emailreader.cron}",
+		name = "${scheduler.emailreader.name}",
+		lockAtMostFor = "${scheduler.emailreader.shedlock-lock-at-most-for}",
+		maximumExecutionTime = "${scheduler.emailreader.maximum-execution-time}")
 	public void getAndProcessEmails() {
 
-		try {
-			RequestId.init();
-
-			LOG.debug("Fetching messages from Emailreader");
-			healthIndicator.resetErrors();
-			emailReaderWorker.getEnabledEmailConfigs()
-				.forEach(config -> emailReaderWorker.getEmailsFromConfig(config)
-					.forEach(email -> {
-						try {
-							emailReaderWorker.processEmail(email, config);
-						} catch (final Exception e) {
-							LOG.error("Error processing email with id: {}", email.getId(), e);
-							healthIndicator.setUnhealthy();
-						}
-					}));
-			if (!healthIndicator.hasErrors()) {
-				healthIndicator.setHealthy();
-			}
-			LOG.debug("Finished fetching from Emailreader");
-		} finally {
-			RequestId.reset();
-		}
+		emailReaderWorker.getEnabledEmailConfigs()
+			.forEach(config -> emailReaderWorker.getEmailsFromConfig(config)
+				.forEach(email -> {
+					try {
+						emailReaderWorker.processEmail(email, config);
+					} catch (final Exception e) {
+						LOG.error("Error processing email with id: {}", email.getId(), e);
+						healthUtility.setHealthIndicatorUnhealthy(jobName, String.format("Error processing email with id: %s", email.getId()));
```

Comment:
Id in health indicator will only be useful when only one email causes exception. I think a more general message is better.